internal/ingress/controller/nginx.go

```diff
@@ -177,14 +184,36 @@ Error loading new template : %v
 
 	n.t = ngxTpl
 
+
 	// TODO: refactor
 	if _, ok := fs.(filesystem.DefaultFs); !ok {
-		watch.NewDummyFileWatcher(tmplPath, onChange)
+		watch.NewDummyFileWatcher(tmplPath, onTemplateChange)
 	} else {
-		_, err = watch.NewFileWatcher(tmplPath, onChange)
-		if err != nil {
-			glog.Fatalf("unexpected error watching template %v: %v", tmplPath, err)
+
+		var watchPath string
+		var watchAction func()
+		var addWatcher func()
+		addWatcher = func() {
```

Comment:
I couldn't work out how to define a function with parameters!? hence why I had to do variables, set them, and invoke the function.  It works, but it doesn't feel right?